javaparser-core/src/main/java/com/github/javaparser/utils/SourceRoot.java

```diff
@@ -259,6 +256,52 @@ public CompilationUnit parse(String startPackage, String filename) {
         }
     }
 
+    private FileVisitResult callback(Path absolutePath, ParserConfiguration configuration, Callback callback) throws IOException {
+        if (!Files.exists(absolutePath)) {
```

Comment:
Should this happen?

Reply:
Yes that can happen. But I see now that the error is either thrown again or properly catched by the invoking methods, so I'll remove this check.